### PR TITLE
doc: nordic: Add instructions for debugging with GDB

### PR DESCRIPTION
**Pull request by carlescufi**
Add a section to the Nordic Segger page that details how to debug a
board with GDB using the Segger GDB server.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

File: doc/tools/nordic_segger.rst [hunk 1]
```diff
@@ -78,6 +78,38 @@ and ``<x>`` is either 1 for nRF51-based boards or 2 for nRF52-based boards
...
 
+Debugging with GDB
+******************
+
+**Note**: The instructions in this section are common to all Operating Systems.
+
+Once the application is flashed into the board, you can opt to debug using GDB
+or try a visual debugger such as Segger Ozone, descibed in :ref:`nordic_segger_ozone`.
+
+* Start the J-Link GDB server on one console:
```
Comment (dbkinder) [inline doc/tools/nordic_segger.rst]: Better is to use numbers for steps, so use `#.` for the list items, and need to indent lines that are part of this list item.

The phrase ``a terminal window`` sounds better than ``one console`` here.

File: doc/tools/nordic_segger.rst [hunk 2]
```diff
@@ -78,6 +78,38 @@ and ``<x>`` is either 1 for nRF51-based boards or 2 for nRF52-based boards
 
 Where ``<x>`` is either 1 for nRF51-based boards or 2 for nRF52-based boards
 
+Debugging with GDB
+******************
+
+**Note**: The instructions in this section are common to all Operating Systems.
+
+Once the application is flashed into the board, you can opt to debug using GDB
```
Comment (dbkinder) [inline doc/tools/nordic_segger.rst]: delete ``opt to``

File: doc/tools/nordic_segger.rst [hunk 3]
```diff
@@ -78,6 +78,38 @@ and ``<x>`` is either 1 for nRF51-based boards or 2 for nRF52-based boards
...
+or try a visual debugger such as Segger Ozone, descibed in :ref:`nordic_segger_ozone`.
+
+* Start the J-Link GDB server on one console:
+
+.. code-block:: console
```
Comment (dbkinder) [inline doc/tools/nordic_segger.rst]: This code block directive and the line below saying "Where..." should be indented by three spaces to be part of the step.

File: doc/tools/nordic_segger.rst [hunk 4]
```diff
@@ -78,6 +78,38 @@ and ``<x>`` is either 1 for nRF51-based boards or 2 for nRF52-based boards
...
+
+   $ JLinkGDBServer -device nrf5<x> -if swd -speed 4000
+
+Where ``<x>`` is either 1 for nRF51-based boards or 2 for nRF52-based boards
+
+* Find ``arm-zephyr-eabi-gdb`` or ``arm-none-eabi-gdb`` depending on whether
```
Comment (dbkinder) [inline doc/tools/nordic_segger.rst]: Next step should use ``#. `` and the lines below (and the code block) should be indented (three spaces) to be part of this step.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by mbolivar** [comment]
I'm confused about why this can't happen as part of "make debug" or "make debugserver". What am I missing?

**Comment by carlescufi** [comment]
@mbolivar It can, we just don't have support for it today and I wanted to document the manual steps until we do.

**Comment by mbolivar** [comment]
> @mbolivar It can, we just don't have support for it today and I wanted to document the manual steps until we do.

I see.

File: doc/tools/nordic_segger.rst [hunk 5]
```diff
@@ -78,6 +78,38 @@ and ``<x>`` is either 1 for nRF51-based boards or 2 for nRF52-based boards
...
 
+Debugging with GDB
+******************
+
+**Note**: The instructions in this section are common to all Operating Systems.
+
+Once the application is flashed into the board, you can debug using GDB or try
+a visual debugger such as Segger Ozone, descibed in :ref:`nordic_segger_ozone`.
+
+# Start the J-Link GDB server on one terminal window:
```
Comment (dbkinder) [inline doc/tools/nordic_segger.rst]: Almost... a numbered list starts with ``#.`` (needs the period) here and below.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by carlescufi** [comment]
Closing, adding this properly via "make debug"

**Pull request closed without merging**